doc/maintainer-guide.md

```diff
@@ -0,0 +1,30 @@
+# Reviewing PRs
+
+Approving a PR means that the reviewer has deemed the PR ready to be merged.
+
+While reviewing
+
+- Check the quality of the code and make sure it conforms to the [coding conventions](coding-convention.md)
+- Test the code with at least InfiniSim or a PineTime
+- Consider and discuss alternative ideas or implementations.
+
+# Merging PRs
+
+Two approvals from core developers is usually required to merge a PR.
+Exceptions include urgent fixes
+and when the author is specified as the maintainer of the code in question.
+Then only a single approval is needed.
+
+Commits that fix code formatting from previous commits must be squashed before merging a PR,
```

Comment:
I think we could ask the author of the PR to make sure the history is clean before we merge the PR, indeed!

Reply:
We haven't had any requirements for commits so far, and this is an idea I had to have just slightly cleaner commits. Requirements for commits is probably something worth discussing, as it relates to #1515, so I think I'll open a new issue for that as well.

Reply:
I think rebase merging gets rid of the merge commits anyway, so a linear history is already implicitly enforced by this guide. And we should definitely enforce a linear history. https://docs.github.com/en/github/administering-a-repository/about-protected-branches#require-linear-history